### ISSUE TITLE
questions regarding _py_proc_get_version 

### ISSUE THREAD

**Issue by stphnlyd** [P403n1x87/austin#92]
1. I wonder why there is this logic https://github.com/P403n1x87/austin/blob/8dce4d29a3c293048ccc319eb48a20b153f638b4/src/py_proc.c#L235 What's the lib_path expected there? And how could it be runnable with a "-V" option? In one of my special environment which is I admit somewhat different from a normal Linux environment, it goes to this place and it thinks lib_path to be some libpython3.Xm.so, which somehow has executable permisson, and runnig the lib file with -V gets a core dump. In contrast in a normal Linux environment, libpython3.X so files are not executable.

2. This place https://github.com/P403n1x87/austin/blob/8dce4d29a3c293048ccc319eb48a20b153f638b4/src/py_proc.c#L248 it's better to find "python" from reverse instead of strstr. In my special environment my libpython so file is under a directory which has "python" in its path, so this logic fails.

**Comment by P403n1x87** [comment]
These are very good questions indeed! This code has always bugged me because it's extremely fragile. That's why I've opened python/cpython#25577. Unfortunately, this will go into 3.11, and for previous versions we're stuck with this unreliable code.

As for the questions:

1. With some of the Python distributions that I've tried (mostly from https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa), when the binary is compiled in a shared library, the library is also executable and gives the Python version with `-V`, like the ordinary binary would. That's why the code assumes that we can treat the library as the actual binary in this case.
2. This coding was based on the assumption that, even if `python` appears in the middle of the path (as a subfolder), it would still contain version information (I think this is generally the case with OSX). Perhaps this could be enhanced by either scanning starting from the end of the string, as you have suggested, or moving forward until the end of the string if previous matching attempts failed.

I'm happy to accept a PR with a solution proposal, else I'll try to take the time to have a look at this as soon as I can.
